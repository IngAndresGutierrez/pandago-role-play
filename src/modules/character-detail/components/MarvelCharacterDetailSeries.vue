<template>
  <v-container class="py-4">
    <v-row justify="center">
      <v-col cols="12">
        <v-carousel v-if="characterSeries.length" height="auto" hide-delimiters>
          <v-carousel-item v-for="(series, index) in characterSeries" :key="index">
            <h3 class="title">{{ series.title }}</h3>
            <v-card-text>{{ series.description }}</v-card-text>
            <v-img :alt="series.title" height="350px" :src="series.thumbnail && getImageFromThumbnail(series.thumbnail)" />
          </v-carousel-item>
        </v-carousel>
        <v-container v-if="loading">
          <v-row justify="center">
            <v-progress-circular color="primary" indeterminate />
          </v-row>
        </v-container>
        <v-alert v-if="!characterSeries.length && !loading" type="info">Este personaje no aparece en ninguna serie.</v-alert>
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
  import { onMounted } from 'vue'
  import { useRoute } from 'vue-router'

  import { Serie } from '../models/series'
  import marvelService from '../services/character'
  import { getImageFromThumbnail } from '../../common/utils/images'

  const route = useRoute()
  const characterSeries = ref<Serie[]>([])
  const loading = ref(false)

  const fetchCharacterDetailSeries = async () => {
    loading.value = true

    if (route.params && 'id' in route.params) {
      const characterId = Number(route.params.id)
      characterSeries.value = await marvelService.getCharacterDetailSeries(characterId)
    }

    loading.value = false
  }

  onMounted(fetchCharacterDetailSeries)

</script>

<style scoped>
.title {
  text-align: center;
}
</style>
