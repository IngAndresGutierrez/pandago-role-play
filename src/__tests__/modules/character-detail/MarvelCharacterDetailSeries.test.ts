import 'vuetify/styles'
import { mount } from '@vue/test-utils'
import { createVuetify } from 'vuetify'
import { describe, expect, test } from 'vitest'

import MarvelCharacterDetailSeries from '@/modules/character-detail/components/MarvelCharacterDetailSeries.vue'

const vuetify = createVuetify()

const mockCharacter = {
  id: 1,
  name: 'Test character',
  thumbnail: {
    path: 'https://mypath.com/images',
    extension: 'jpg',
  },
}

describe('MarvelCharacterDetailSeries.vue', () => {
  test('renders correctly', () => {
    const wrapper = mount(MarvelCharacterDetailSeries, {
      global: {
        plugins: [vuetify],
      },
      props: {
        character: mockCharacter,
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
