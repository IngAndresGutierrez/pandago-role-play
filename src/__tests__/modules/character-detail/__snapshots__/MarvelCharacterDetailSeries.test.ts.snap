// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MarvelCharacterDetailSeries.vue > renders correctly 1`] = `
<div
  character="[object Object]"
  class="v-container v-locale--is-ltr py-4"
  data-v-43de85d1=""
>
  <div
    class="v-row justify-center"
    data-v-43de85d1=""
  >
    <div
      class="v-col v-col-12"
      data-v-43de85d1=""
    >
      <!--v-if-->
      <!--v-if-->
      <div
        class="v-alert v-theme--light bg-info v-alert--density-default v-alert--variant-flat"
        data-v-43de85d1=""
        role="alert"
      >
        
        <!---->
        <span
          class="v-alert__underlay"
        />
        
        <!---->
        <div
          class="v-alert__prepend"
        >
          <i
            aria-hidden="true"
            class="mdi-information mdi v-icon notranslate v-theme--light"
            density="default"
            style="font-size: 28px; height: 28px; width: 28px;"
          />
        </div>
        <div
          class="v-alert__content"
        >
          <!---->
          <!---->
          
          Este personaje no aparece en ninguna serie.
          
        </div>
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;
